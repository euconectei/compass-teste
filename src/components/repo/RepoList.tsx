import { RepoType } from '../../types/RepoType';

type RepoListType = {
  repos: RepoType[];
};

const RepoList = ({ repos }: RepoListType) => {
  if (!repos?.length) {
    return (
      <div className='alert alert-info'>Nenhum reposit√≥rio encontrado.</div>
    );
  }

  return (
    <ul className='nav flex-column'>
      {repos.map((repo, i) => (
        <li key={`repo-${i}`} className='nav-item'>
          <a href={repo.html_url} className='nav-link'>
            {repo.name}
          </a>
        </li>
      ))}
    </ul>
  );
};

export default RepoList;
