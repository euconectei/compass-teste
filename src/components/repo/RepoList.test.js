import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { reposMock } from '../../types/__mocks__/repos.mock';
import RepoList from './RepoList';

describe('Testing UserDisplay Component', () => {
  it('shold render the component', () => {
    const component = renderer.create(<RepoList repos={reposMock} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with an empty repos', () => {
    const { getByText } = render(<RepoList repos={[]} />);
    expect(getByText('Nenhum repositÃ³rio encontrado.')).toBeInTheDocument();
  });

  it('should render the component with repos', () => {
    const { getByText } = render(<RepoList repos={reposMock} />);
    expect(getByText('AcessibilidadeUsabilidade')).toBeInTheDocument();
  });
});
