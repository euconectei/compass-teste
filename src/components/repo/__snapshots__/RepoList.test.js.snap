// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserDisplay Component shold render the component 1`] = `
<ul
  className="nav flex-column"
>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/AcessibilidadeUsabilidade"
    >
      AcessibilidadeUsabilidade
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/autocompleteTagchat"
    >
      autocompleteTagchat
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/carga-instalada"
    >
      carga-instalada
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/ci-cd-travis-hostgator"
    >
      ci-cd-travis-hostgator
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/classificacao-conversa"
    >
      classificacao-conversa
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/codecopy"
    >
      codecopy
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/condo"
    >
      condo
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/Corporizacao-ABCiber"
    >
      Corporizacao-ABCiber
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/create-react-app"
    >
      create-react-app
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/demo-site-awesome-react-links"
    >
      demo-site-awesome-react-links
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/euconectei.github.io"
    >
      euconectei.github.io
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/gestao-de-convidados"
    >
      gestao-de-convidados
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/gift-manager"
    >
      gift-manager
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/git-point"
    >
      git-point
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/Gulp_init"
    >
      Gulp_init
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/Hacktoberfest-2018-First-Pull-Request"
    >
      Hacktoberfest-2018-First-Pull-Request
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/hacktoberfest2018"
    >
      hacktoberfest2018
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/hamb-menu"
    >
      hamb-menu
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/Make-a-Pull-Request"
    >
      Make-a-Pull-Request
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/mci"
    >
      mci
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/mean-exemplo"
    >
      mean-exemplo
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/Meny"
    >
      Meny
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/pagamento-react"
    >
      pagamento-react
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/phonegap-cloud9"
    >
      phonegap-cloud9
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/Projetos-Colaborativos"
    >
      Projetos-Colaborativos
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/resrc"
    >
      resrc
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/reveal.js"
    >
      reveal.js
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/setup_config"
    >
      setup_config
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/site"
    >
      site
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="https://github.com/euconectei/socorrovovo"
    >
      socorrovovo
    </a>
  </li>
</ul>
`;
